jcl/src/java.base/share/classes/java/lang/invoke/MethodTypeHelper.java

```diff
@@ -218,13 +218,21 @@ static final int parseIntoClass(char[] signature, int index, ArrayList<Class<?>>
 		char current = signature[index];
 		Class<?> c;
 
-		if ((current == 'L') || (current == '[')) {
+		if ((current == 'L') || (current == '[')
```

Comment:
You may want to modify `nonPrimitiveClassFromString()` above as well.

Reply:
updated in f180a75